### ISSUE TITLE
Error applying patch created by `make_patch`

### ISSUE THREAD

**Issue by trvrm** [stefankoegl/python-json-patch#54]
applying a simple patch created by diffing a list containing a dict:

``` python
        src = [{}]
        dst = [{'key': ''}]
        patch = jsonpatch.make_patch(src, dst)
        res = patch.apply(src)
```

causes

```
jsonpatch.JsonPatchConflict: can't replace non-existent object 'key'
```

**Comment by trvrm** [comment]
The patch seems wrong:

```python
print(patch.patch)
```

```python
[{'value': '', 'path': '/0/key', 'op': 'replace'}]
```

Whereas if I just try to diff the inner dict:

```python
jsonpatch.make_patch(src[0],dst[0]).patch
```

I get
```python
[{'op': 'add', 'path': '/key', 'value': ''}]
```

Which is what I'd expect.

**Comment by trvrm** [comment]
This is a regression - it doesn't happen in versions prior to https://github.com/stefankoegl/python-json-patch/commit/a33021bf5a87350abc225a15c2a12880d88ed383

**Comment by stefankoegl** [comment]
This should have been resolved by #59. I will release a new version soon.
